pybotx/models/message/mentions.py

```diff
@@ -16,52 +16,101 @@
 
 
 @dataclass
-class Mention:
-    type: MentionTypes
-    entity_id: Optional[UUID] = None
-    name: Optional[str] = None
+class BaseMentionEntity:
+    entity_id: UUID
+    name: Optional[str]
 
-    def __str__(self) -> str:
+    def _build_embed_mention(self, mention_type: MentionTypes) -> str:
         name = self.name or ""
         entity_id = self.entity_id or ""
-        mention_type = self.type.value
-        return f"<embed_mention>{mention_type}:{entity_id}:{name}</embed_mention>"
+        return f"<embed_mention>{mention_type.value}:{entity_id}:{name}</embed_mention>"
+
+
+@dataclass
+class MentionUser(BaseMentionEntity):
+    type: Literal[MentionTypes.USER]
+
+    def __str__(self) -> str:
+        return self._build_embed_mention(self.type)
+
+
+@dataclass
+class MentionContact(BaseMentionEntity):
+    type: Literal[MentionTypes.CONTACT]
+
+    def __str__(self) -> str:
+        return self._build_embed_mention(self.type)
+
+
+@dataclass
+class MentionChat(BaseMentionEntity):
+    type: Literal[MentionTypes.CHAT]
+
+    def __str__(self) -> str:
+        return self._build_embed_mention(self.type)
+
+
+@dataclass
+class MentionChannel(BaseMentionEntity):
+    type: Literal[MentionTypes.CHANNEL]
+
+    def __str__(self) -> str:
+        return self._build_embed_mention(self.type)
+
+
+@dataclass
+class MentionAll:
+    type: Literal[MentionTypes.ALL]
+
+    def __str__(self) -> str:
+        return f"<embed_mention>{self.type.value}::</embed_mention>"  # noqa: WPS237
```

Comment:
Also use `_build_embed_mention` here. May be pass `entity_id` and `name` through args?

Reply:
Fix